File: src/main/java/BuildingType.java
package main.java;

public enum BuildingType{
    APARTMENT(0),
    VILLA(1);

    private final int value;

    private BuildingType(int value){
        this.value = value;
    }

    @Override
    public String toString() {
        if(value == 0)
            return "آپارتمان";
        else
            return "ویلایی";
    }
}

File: back-end-prototype/domain/buildingType.js
'use strict';
var debug = require('debug')('khanebedoosh:domain');

const BuildingType = {
    VILLA: 'VILLA',
    APARTMENT: 'APARTMENT'
};

var parse = int => {
    switch (int) {
        case 0:
        case 'VILLA':
        case 'ویلایی':
        case 'ويلايي':
            return BuildingType.VILLA;
        case 1:
        case 'APARTMENT':
        case 'آپارتمان':
        case 'اپارتمان':
            return BuildingType.APARTMENT;
        default:
            throw 'Invalid BuildingType';
    }
};

module.exports = {
    BuildingType: Object.freeze(BuildingType),
    toBuildingType: parse
};
